### PR TITLE
Makefile update

### PR DESCRIPTION
**Pull request by Jim3Things**
There are three main points to this change:
- The proto files have been moved back to simulation, and now publishes the typescript and C# proto files to the common pkg directory tree
- The observer project is incorporated into the common build, via a sub-makefile
- The makefile has removed much of the replication in file definitions, and correctly tracks dependencies, such that repeated build commands do nothing...

Thanks to Mike for most of the observer makefile and quite a few parts of the global makefile changes.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,102 +1,60 @@
...
+	simulation/pkg/protos/inventory/target.proto \
+	simulation/pkg/protos/workload/actual.proto \
+    simulation/pkg/protos/workload/external.proto \
+    simulation/pkg/protos/workload/internal.proto \
+    simulation/pkg/protos/workload/target.proto \
+    simulation/pkg/protos/services/inventory.proto \
+    simulation/pkg/protos/services/monitor.proto \
+    simulation/pkg/protos/services/requests.proto \
+    simulation/pkg/protos/services/stepper.proto \
+    simulation/pkg/protos/services/traceSink.proto
```
Comment (MikeCJohnsonUk) [inline Makefile:38]: spaces vs. tabs?

File: Makefile [hunk 2]
```diff
@@ -203,33 +161,41 @@ SRC_WEBSERVER = \
...
+    $(KIT_BUILD)/controllerd.exe \
+    $(KIT_BUILD)/inventoryd.exe \
+    $(KIT_BUILD)/sim_supportd.exe \
+    $(KIT_BUILD)/web_server.exe
 
 VERSION_MARKER = \
     simulation/pkg/version/generated.go \
-    simulation/pkg/version/version_stamp.md
+    simulation/pkg/version/version_stamp.md \
+	simulation/pkg/version/readme.md
```
Comment (MikeCJohnsonUk) [inline Makefile:190]: spaces vs. tabs?

File: Makefile [hunk 3]
```diff
@@ -203,33 +161,41 @@ SRC_WEBSERVER = \
...
-    simulation/deployments/Deploy.cmd \
-    simulation/deployments/StartAll.cmd \
-    simulation/deployments/StartCloudChamber.cmd \
-    simulation/deployments/StartEtcd.cmd \
-    simulation/deployments/MonitorEtcd.cmd
 
+INSTALL_KIT = $(SERVICES) $(ARTIFACTS) $(OBSERVER)
 
 
 INSTALL_KIT = $(SERVICES) $(ARTIFACTS)
```
Comment (MikeCJohnsonUk) [inline Makefile:196]: Do we get both definitions?

File: Makefile [hunk 4]
```diff
@@ -293,91 +273,113 @@ install_clean:
...
+	$(UT_GO) $(PROJECT)/simulation/internal/tracing/exporters
+	$(MAKE) -C $(OBSERVER_UI) test
 
 
 .PHONY : clean
 
 clean:
-	$(RM) $(SERVICES) $(ARTIFACTS) $(PROTO_GEN_FILES) $(PROTO_CS_GEN_FILES) $(PROTO_TS_GEN_FILES) $(VERSION_MARKER)
+	$(RM) $(SERVICES) $(ARTIFACTS) $(PROTO_GEN_FILES) $(PROTO_CS_GEN_FILES) $(PROTO_TS_GEN_FILES) $(EXPORT_CS_GEN_FILES) $(EXPORT_TSX_FILES) $(VERSION_MARKER)
+	$(MAKE) -C $(OBSERVER_UI) clean
```
Comment (MikeCJohnsonUk) [inline Makefile:305]: Do we also need an 

$(RM-RECURSIVE) $(KIT_BUILD)

as part of the recipe?

File: Makefile [hunk 5]
```diff
@@ -293,91 +273,113 @@ install_clean:
...
-	$(CP) $(PROJECT)/$< $(PROJECT)/$@
-
-simulation/deployments/StartEtcd.cmd: simulation/scripts/StartEtcd.cmd
-	$(CP) $(PROJECT)/$< $(PROJECT)/$@
-
-simulation/deployments/StartAll.cmd : simulation/scripts/StartAll.cmd
-	$(CP) $(PROJECT)/$< $(PROJECT)/$@
-
-simulation/deployments/StartCloudChamber.cmd : simulation/scripts/StartCloudChamber.cmd
+simulation/deployments/inventory.yaml: simulation/configs/Inventory.yaml
```
Comment (MikeCJohnsonUk) [inline Makefile:381]: You could probably do the same thing here as for the .cmd files

simulation/deployments/%.yaml: simulation/scripts/%.yaml
	$(CP) $(PROJECT)/$< $(PROJECT)/$@

Comment (Jim3Things) [reply Makefile:381]: I did try that at first, but ran into the Inventory vs inventory issue.  The reader code expects 'inventory', so we really need to have that in the deployment directory, and the checked-in configs file is 'Inventory'.  Ugh.

File: Makefile (hunk 1 above)
Comment (Jim3Things) [reply Makefile]: yeah... I fixed these and the others I found.

File: Makefile (hunk 2 above)
Comment (Jim3Things) [reply Makefile]: yep.  fixed.

File: Makefile (hunk 3 above)
Comment (Jim3Things) [reply Makefile]: Good catch.  Missed that.  I've removed the second one.

File: Makefile (hunk 4 above)
Comment (Jim3Things) [reply Makefile:302]: Good idea.  I've added that.  I also added the cleanup on install that fixes issue #301

File: Makefile (hunk 5 above)
Comment (MikeCJohnsonUk) [reply Makefile:386]: ok.

File: Makefile (hunk 1 above)
Comment (MikeCJohnsonUk) [reply Makefile]: Ok.

File: Makefile (hunk 2 above)
Comment (MikeCJohnsonUk) [reply Makefile]: Ok.

File: Makefile (hunk 3 above)
Comment (MikeCJohnsonUk) [reply Makefile]: Ok.

File: Makefile (hunk 4 above)
Comment (MikeCJohnsonUk) [reply Makefile:302]: Ok.

**Review by MikeCJohnsonUk — APPROVED** [review]

**Pull request merged**